SkipList/SkipList.Test/SkipListTests.cs

```diff
@@ -0,0 +1,254 @@
+namespace SkipListTests;
+
+using MySkipList;
+
+public class SkipListTests
+{
+    SkipList<int> skiplist;
+
+    [SetUp]
+    public void SetUp()
+        => skiplist = [];
+
+    [Test]
+    public void CountWithAdd_ShouldReturnExpectedResult()
+    {
+        Array.ForEach(new int[] { 1, 7, 4, 5, 6 }, skiplist.Add);
+
+
+        var expectedResultAfterAdd = 5;
+        Assert.That(skiplist, Has.Count.EqualTo(expectedResultAfterAdd));
+    }
+
+    [Test]
+    public void CountWithRemove_ShouldReturnExpectedResult()
+    {
+        Array.ForEach(new int[] { 1, 7, 4, 5, 6 }, skiplist.Add);
+
+
+        skiplist.Remove(4);
+        skiplist.Remove(10);
+        skiplist.Remove(1);
+        skiplist.Remove(-1);
+
+
+        var expectedResultAfterRemove = 3;
+        Assert.That(skiplist, Has.Count.EqualTo(expectedResultAfterRemove));
+    }
+
+    [Test]
+    public void IsReadOnly_ShouldReturnFalse()
+        => Assert.That(!skiplist.IsReadOnly);
+
+    [Test]
+    public void IndexerWithAddAndRemove_WithCorrectIndex_ShouldReturnExpectedResult()
+    {
+        Array.ForEach(new int[] { 1, 7, 4, 5, 12 }, skiplist.Add);
+
+
+        skiplist.Remove(4);
+        skiplist.Remove(7);
+        skiplist.Remove(10);
+        skiplist.Remove(-1);
+
+
+        var expectedArray = new int[] { 1, 5, 12 };
+        for (int i = 0; i < expectedArray.Length; ++i)
+        {
+            Assert.That(skiplist[i], Is.EqualTo(expectedArray[i]));
+        }
+    }
+
+    [Test]
+    public void SetByIndex_ShouldThrowsNotSupportedException()
+        => Assert.Throws<NotSupportedException>(() => skiplist[0] = 10);
+
+    [Test]
+    public void GetByIndex_WithIncorrectIndex_ShouldThrowArgumentOutOfRangeException()
+    {
+        skiplist.Add(1);
+        var value = 0;
+
+
+        Assert.Throws<ArgumentOutOfRangeException>(() => value = skiplist[1]);
+        Assert.Throws<ArgumentOutOfRangeException>(() => value = skiplist[-1]);
+    }
+
+    [Test]
+    public void ContainsWithAddAndRemove_WithCorrectInput_ShouldReturnExpectedResult()
+    {
+        Assert.That(skiplist, Does.Not.Contain(10));
+
+
+        Array.ForEach(new int[] { 9, 7, 5, 12 }, skiplist.Add);
+        
+
+        skiplist.Remove(9);
+        skiplist.Remove(7);
+
+
+        Assert.That(skiplist, Does.Contain(5));
+        Assert.That(skiplist, Does.Contain(12));

```

Comment:
В паттерне ААА всего три части, а тут их получилось аж 4, выглядит странно...
```suggestion
        Assert.That(skiplist, Does.Not.Contain(10));


        Array.ForEach(new int[] { 9, 7, 5, 12 }, skiplist.Add);

        skiplist.Remove(9);
        skiplist.Remove(7);


        Assert.That(skiplist, Does.Contain(5));
        Assert.That(skiplist, Does.Contain(12));

```

Reply:
Просто первая проверка как бы является частью Assert, а не Arrange. По хорошему нужно было отдельный тест наверное сделать, но я поленился

Reply:
тоже правда